sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -191,6 +198,11 @@ class BulkWriter(container: CosmosAsyncContainer,
     )
   }
 
+  def isSuccessStatusCode(itemResponse: CosmosBulkItemResponse, itemOperation: CosmosItemOperation): Boolean = {
+    (itemResponse.isSuccessStatusCode
+     || (itemOperation.getOperationType == CosmosItemOperationType.PATCH && itemResponse.getStatusCode == HttpConstants.StatusCodes.NOT_MODIFIED))
```

Comment:
Just curious to know where else do we return 304 status code - `NOT_MODIFIED` ?

Reply:
Because in Java SDK - 304 is always success, I think: 

This is from `HttpTransportClient.java`
```
// If the status code is < 300 or 304 NotModified (we treat not modified as success) then it means that it's a success code and shouldn't throw.
        if (response.statusCode() < HttpConstants.StatusCodes.MINIMUM_STATUSCODE_AS_ERROR_GATEWAY ||
                response.statusCode() == HttpConstants.StatusCodes.NOT_MODIFIED) {
            return ResponseUtils.toStoreResponse(response, httpRequest);
        }
        else {
            return this.createErrorResponseFromHttpResponse(resourceAddress, activityId, httpRequest, response);
        }
```

Reply:
for 304, so far I know two places:
1. for patch: when there are no operations being included in the patch operation
2. for change feed: when trying to check any new recent change, if there is no new change, it will return not_modified

Reply:
and I think I will revert this change back as:
For now, we start from more strict rules: throw exception if empty operations scenario happens.